module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp

```diff
@@ -305,54 +327,164 @@ namespace bsp::battery_charger
+        }
 
+        batteryRetval verifySavedConfigurationFile(const std::vector<Register> &dataForVerification)
+        {
+            std::ifstream readFile(cfgFile.c_str(), std::ios::binary | std::ios::in);
+            if (!readFile.is_open()) {
+                LOG_WARN("Configuration file [%s] could not be opened.", cfgFile.c_str());
+                return batteryRetval::ChargerError;
+            }
+            if (dataForVerification.size() != registersToStore) {
+                LOG_ERROR("Wrong data size.");
+                return batteryRetval::ChargerError;
+            }
+            std::uint16_t savedRegValue;
+            for (unsigned int i = 0; i < registersToStore; ++i) {
+                readFile.read(reinterpret_cast<char *>(&savedRegValue), sizeof(savedRegValue));
+                if (savedRegValue != dataForVerification[i]) {
+                    LOG_ERROR("Store verification 0x%x failed. Value read from file: %d from REG: %d",
+                              i,
+                              savedRegValue,
+                              dataForVerification[i]);
+                    readFile.close();
+                    return batteryRetval::ChargerError;
+                }
+            }
+            readFile.close();
             return batteryRetval::OK;
         }
 
-        batteryRetval loadConfiguration()
+        batteryRetval readConfigurationFile(std::vector<Register> &readData)
         {
             std::ifstream file(cfgFile.c_str(), std::ios::binary | std::ios::in);
             if (!file.is_open()) {
                 LOG_WARN("Configuration file [%s] could not be opened. Loading initial configuration.",
                          cfgFile.c_str());
-                if (configureFuelGaugeBatteryModel() == batteryRetval::OK) {
-                    storeConfiguration();
-                    resetFuelGaugeModel();
-                    return batteryRetval::OK;
-                }
-                else {
-                    return batteryRetval::ChargerError;
-                }
+                return batteryRetval::ChargerError;
+            }
+            if (readData.size() != registersToStore) {
+                LOG_ERROR("Wrong data size.");
+                return batteryRetval::ChargerError;
             }
-
             std::uint16_t regVal;
             for (auto i = 0; i < registersToStore; ++i) {
-                file.read(reinterpret_cast<char *>(&regVal), sizeof(std::uint16_t));
-                if (fuelGaugeWrite(static_cast<Registers>(i), regVal) != kStatus_Success) {
+                file.read(reinterpret_cast<char *>(&regVal), sizeof(regVal));
+                readData[i] = regVal;
+            }
+            file.close();
+            if (!bsp::battery_charger::utils::isCorrectChecksumFromConfigFile(cfgFile.c_str())) {
+                LOG_WARN("Configuration file is corrupted. Loading initial configuration.");
+                return batteryRetval::ChargerError;
```

Comment:
just wondering, wouldn't it be more verbose if it was a `ChecksumError` or something similiar? Right now all errors are charger errors, so it could be just an error :D

Reply:
Very true. It would be nice t drop all these logs altogether and just use the correct ret values. Printing logs should  not be the responsibility of this module in my opinion.

Reply:
done